Make this project POSIX Compliant.

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
<!--- Tell us what should happen -->
Make this project POSIX Compliant.

* [ShellCheck &ndash; shell script analysis tool ](https://www.shellcheck.net/);
* [24 Bashism To Avoid for POSIX-Compliant Shell Scripts](https://betterprogramming.pub/24-bashism-to-avoid-for-posix-compliant-shell-scripts-8e7c09e0f49a);
* [POSIX compliant shell scripting - HLRS Platforms ](https://kb.hlrs.de/platforms/index.php/POSIX_compliant_shell_scripting)
* [GitHub - koalaman/shellcheck: ShellCheck, a static analysis tool for shell scripts ](https://github.com/koalaman/shellcheck)

## Current Behavior

#### For `tsearch`
<!--- Tell us what happens instead of the expected behavior -->
```sh

In tsearch line 24:
lang="en"
^--^ SC2034: lang appears unused. Verify use (or export if used externally).


In tsearch line 49:
            [a-zA-Z0-9.~_-]) printf "$c" ;;
                                    ^--^ SC2059: Don't use variables in the printf format string. Use printf '..%s..' "$foo".


In tsearch line 72:
    case "${option}" in
    ^-- SC2220: Invalid flags are not handled. Add a *) case.


In tsearch line 88:
    $command $(get_query $engine)
             ^------------------^ SC2046: Quote this to prevent word splitting.
                         ^-----^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
    $command $(get_query "$engine")

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- lang appears unused. Verify use (...
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
  https://www.shellcheck.net/wiki/SC2220 -- Invalid flags are not handled. Ad...

```

#### For `dmenu_tsearch`

```sh
In dmenu_tsearch line 3:
CONFIG=/etc/tinytools/params.yaml
^-- [SC2034](https://github.com/koalaman/shellcheck/wiki/SC2034) (warning): CONFIG appears unused. Verify use (or export if used externally).
 
In dmenu_tsearch line 7:
CLIP_OPTION=$(echo $(xclip -selection c -o))
                   ^-- [SC2046](https://github.com/koalaman/shellcheck/wiki/SC2046) (warning): Quote this to prevent word splitting.
                   ^-- [SC2005](https://github.com/koalaman/shellcheck/wiki/SC2005) (style): Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
```
## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1.
2.
3.
4.

## Context (Environment)
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

<!--- Provide a general summary of the issue in the Title above -->

## Detailed Description
<!--- Provide a detailed description of the change or addition you are proposing -->

## Possible Implementation
<!--- Not obligatory, but suggest an idea for implementing addition or change -->

--- a/dmenu_tsearch
+++ b/dmenu_tsearch
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 ENGINE_OPTION=">> CHANGE ENGINE"
 PROGRAM_OPTION=">> CHANGE PROGRAM"
@@ -18,11 +18,11 @@ if [ -n "$1" ]; then
 fi
 
 main(){
-    while [ $IS_SPECIAL_INPUT == 1 ]; do
-        TERM=$(echo -e "$CLIP_OPTION\n$ENGINE_OPTION\n$PROGRAM_OPTION" | dmenu -p "$PROMPT:" -i)
+    while [ $IS_SPECIAL_INPUT = 1 ]; do
+        TERM=$(printf '%s\n%s\n%s' "$CLIP_OPTION" "$ENGINE_OPTION" "$PROGRAM_OPTION" | dmenu -p "$PROMPT:" -i)
         case "$TERM" in
             "$ENGINE_OPTION")
-                SELECTED_ENGINE=$(echo -e "$ENGINE_CHOICES" | dmenu -p "Engine:" -i -l 10)
+                SELECTED_ENGINE=$(printf '%s' "$ENGINE_CHOICES" | dmenu -p "Engine:" -i -l 10)
                 if [ -n "$SELECTED_ENGINE" ]; then
                     ENGINE="$SELECTED_ENGINE"
                     PROMPT="$ENGINE"
@@ -31,7 +31,7 @@ main(){
                 IS_SPECIAL_INPUT=1
             ;;
             "$PROGRAM_OPTION")
-                SELECTED_PROGRAM=$(echo -e "$PROGRAM_CHOICES" | dmenu -p "Program:" -i)
+                SELECTED_PROGRAM=$(printf '%s' "$PROGRAM_CHOICES" | dmenu -p "Program:" -i)
                 if [ -n "$SELECTED_PROGRAM" ]; then
                     PROGRAM="$SELECTED_PROGRAM"
                     COMMAND="$COMMAND -p $PROGRAM"

--- a/dmenu_tsearch
+++ b/dmenu_tsearch
@@ -1,10 +1,8 @@
 #!/bin/bash
 
-CONFIG=/etc/tinytools/params.yaml
-
 ENGINE_OPTION=">> CHANGE ENGINE"
 PROGRAM_OPTION=">> CHANGE PROGRAM"
-CLIP_OPTION=$(echo $(xclip -selection c -o))
+CLIP_OPTION="$(xclip -selection c -o | tr '\n' ' ')"
 
 ENGINE_CHOICES=$(tsearch -l engine)
 PROGRAM_CHOICES=$(tsearch -l program)

--- a/tsearch
+++ b/tsearch
@@ -21,7 +21,6 @@ search_params="tyaml /etc/tinytools/params.yaml"
 
 program=$($search_params -v default.program)
 engine=$($search_params -v default.engine)
-lang="en"
 
 usage(){
    echo -e "\033[1musage:\033[0m"
@@ -45,7 +44,7 @@ url_encode(){
     for (( i = 0; i < length; i++ )); do
         local c="${1:i:1}"
         case $c in
-            [a-zA-Z0-9.~_-]) printf "$c" ;;
+            [a-zA-Z0-9.~_-]) printf '%s' "$c" ;;
             ' ') printf "+" ;;
             *) printf '%%%02X' "'$c" ;;
         esac
@@ -74,6 +73,7 @@ do
         t) term=$(url_encode "${OPTARG}");;
         h) usage;;
         l) list "${OPTARG}";;
+        *) usage; exit 1 ;;
     esac
 done
 
@@ -84,8 +84,7 @@ if [ -n "$term" ]; then
     }
 
     command=$($search_params -v program."$program".command)
-    $command $(get_query $engine)
+    $command "$(get_query "$engine")"
 else
     echo "no term declared"
 fi
-

--- a/tsearch
+++ b/tsearch
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #This file is part of the TinyTools distribution (https://github.com/Calebe94/TinyTools).
 #Copyright (C) 2021 TinyTools
@@ -23,31 +23,33 @@ program=$($search_params -v default.program)
 engine=$($search_params -v default.engine)
 
 usage(){
-   echo -e "\033[1musage:\033[0m"
-   echo -e "tsearch -t \"term\" [-e] engine [-p] program"
-   echo -e "\033[1moptions:\033[0m"
-   echo -e "\033[1m-t:\033[0m term to search"
-   echo -e "\033[1m-e:\033[0m set search engine"
-   echo -e "\033[1m-p:\033[0m set program to launch"
-   echo -e  "\033[1m-l <engine|program>:\033[0m list engines/programs available"
-   echo -e "\033[1mexample:\033[0m"
-   echo -e "\033[1m$\033[0m tsearch -t \"free software\" -e duckduckgo -p firefox"
-   echo -e "- all search parameters are located in /etc/tinytools/params.yaml"
+   printf "\033[1musage:\033[0m\n"
+   printf "tsearch -t \"term\" [-e] engine [-p] program\n"
+   printf "\033[1moptions:\033[0m\n"
+   printf "\033[1m-t:\033[0m term to search\n"
+   printf "\033[1m-e:\033[0m set search engine\n"
+   printf "\033[1m-p:\033[0m set program to launch\n"
+   printf "\033[1m-l <engine|program>:\033[0m list engines/programs available\n"
+   printf "\033[1mexample:\033[0m\n"
+   printf "\033[1m$\033[0m tsearch -t \"free software\" -e duckduckgo -p firefox\n"
+   printf -- "- all search parameters are located in /etc/tinytools/params.yaml\n"
    exit 0
 }
 
 url_encode(){
     old_lc_collate=$LC_COLLATE
     LC_COLLATE=C
-
-    local length="${#1}"
-    for (( i = 0; i < length; i++ )); do
-        local c="${1:i:1}"
+    i=1
+    length="${#1}"
+    while [ "$i" -le "$length" ]
+    do
+        c="$(printf '%s' "$1" | cut -c $i)"
         case $c in
             [a-zA-Z0-9.~_-]) printf '%s' "$c" ;;
             ' ') printf "+" ;;
             *) printf '%%%02X' "'$c" ;;
         esac
+        i=$((i+1))
     done
 
     LC_COLLATE=$old_lc_collate
@@ -57,10 +59,10 @@ list()
 {
     case "$1" in
         engine|program)
-            echo -e "$($search_params -k "$1")"
+            printf '%s\n' "$($search_params -k "$1")"
             ;;
         default*)
-            echo -e "$($search_params -v "$1")"
+            printf '%s\n' "$($search_params -v "$1")"
     esac
     exit 0
 }
@@ -80,11 +82,11 @@ done
 if [ -n "$term" ]; then
     get_query(){
         query=$($search_params -v engine."$1".query)
-        echo "${query/TERM/$term}"
+        printf '%s' "$query" | sed "s/TERM/$term/g"
     }
 
     command=$($search_params -v program."$program".command)
     $command "$(get_query "$engine")"
 else
-    echo "no term declared"
+    printf '%s' "no term declared"
 fi

